compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -2271,25 +2272,7 @@ private void addErrorTypesToSet(BType returnType, LinkedHashSet<BType> errorType
     }
 
     private boolean hasNonErrorType(BType returnType) {
-        if (returnType == null) {
-            return false;
-        }
-
-        BType effType = Types.getImpliedType(types.getTypeWithEffectiveIntersectionTypes(returnType));
-        if (effType.tag == TypeTags.ERROR) {
-            return false;
-        }
-
-        if (effType.tag == TypeTags.UNION) {
-            for (BType memberType : ((BUnionType) returnType).getMemberTypes()) {
-                if (hasNonErrorType(memberType)) {
-                    return true;
-                }
-            }
-            return false;
-        }
-
-        return true;
+        return !Core.isEmpty(types.typeCtx(), Core.diff(returnType.semType(), PredefinedType.ERROR));
```

Comment:
Isn't this same as `!Core.isSubtyp(returnType.semType(), PredefinedType.ERROR))`?  Also in the previous code we had handled the case where `returnType` could be null don't we need to handle it here a well?

Reply:
Fixed the former. I think the null check we had previously is redundant/obsolete. No test for that. Hence, removed.
 

<img width="449" alt="Screenshot 2024-11-11 at 18 15 07" src="https://github.com/user-attachments/assets/29f85f7e-568a-4fcb-85ac-9efc3304044f">